video/out/drm_atomic.c

```diff
@@ -458,3 +495,85 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
 
     return (ret == 0);
 }
+
+static inline uint16_t clamp16(float x)
+{
+    return (uint16_t)MPCLAMP(x, 0.0, 65535.0);
+}
+
+static inline uint16_t clamp16_round(float x)
+{
+    return clamp16(roundf(x * 50000.0));
+}
+
+void drm_create_hdr_metadata(struct drm_atomic_context *ctx, struct drm_hdr *hdr_data, bool reset)
+{
+#if HAVE_DRM_HDR
+    struct hdr_output_metadata metadata = { .metadata_type = DRM_STATIC_METADATA_TYPE1,
+                                            .hdmi_metadata_type1 = {0} };
```

Comment:
oh yeah btw this would be a good idea:
```suggestion
    struct hdr_output_metadata metadata = {0};
    metadata.metadata_type = DRM_STATIC_METADATA_TYPE1;
```